yktrace/src/hwt/mapper.rs

```diff
@@ -134,59 +130,66 @@ impl<'a> HWTMapper<'a> {
         ret
     }
 
-    /// Obtain the next mapped block.
+    /// Map the *machine* blocks of the specified trace into LLVM IR blocks.
     ///
-    /// `last_yielded` must be a reference to the last successfully mapped block returned by the
-    /// iterator (or `None` if this is the first time you are calling `next`).
-    pub fn next(
+    /// The returned trace is guaranteed to have the following properties:
+    ///
+    ///  - The returned trace will never start with an unmappable code block. It is a given that
```

Comment:
I eventually worked out what this meant, but can we try simplifying the explanation to make it clearer?

Reply:
Try 42e670140381fcf31b31ac5cc91ed93f53697fc1